packages/popover2/src/popover2SharedProps.ts

```diff
@@ -283,6 +283,13 @@ export interface IPopover2SharedProps<TProps> extends OverlayableProps, Props {
      */
     targetTagName?: keyof JSX.IntrinsicElements;
 
+    /**
+     * HTML props for the target element.
+     *
+     * This prop is mutually exclusive with the `renderTarget` API.
+     */
+    targetWrapperProps?: React.HTMLProps<HTMLSpanElement> | React.HTMLProps<HTMLDivElement>;
```

Comment:
```suggestion
    targetProps?: TProps;
```

Reply:
OK done, please review.

In the case of a `renderTarget`, `targetProps` props will also be appliedto the `renderTarget`, so I made the warning basically say that they will be applied, but that it doesn't really make sense to do it that way vs. just applying them in your `renderTarget`.

Reply:
@adidahiya  I'm not sure why CI is failing, is `TProps` not the best type for `targetProps` here maybe?

Reply:
`TProps` is the correct type to use here, but adding this constraint exposed some type issues with the way I had designed Popover2. It required adding a stricter typedef to `validateProps`, which further exposed some complex issues with the `renderTarget` API and the inferred value of `TProps`. I think I've solved those issues in the commits I just pushed to this branch.